src/components/RichText.tsx

```diff
@@ -43,215 +29,114 @@ export function RichText({
-    }
+    return renderSimpleText(text, selectable, testID, numberOfLines, plainStyles, emojiMultiplier);
+  }
+
+  const els = buildTextElements(richText, disableLinks, enableTags, authorHandle, interactiveStyles, onLinkPress);
+
+  return (
+    <Text selectable={selectable} testID={testID} style={plainStyles} numberOfLines={numberOfLines}>
+      {els}
+    </Text>
+  );
+}
+
+function renderSimpleText(text, selectable, testID, numberOfLines, plainStyles, emojiMultiplier) {
+  if (isOnlyEmoji(text)) {
+    const fontSize = (flattenedStyle.fontSize ?? a.text_sm.fontSize) * emojiMultiplier;
     return (
-      <Text
-        selectable={selectable}
-        testID={testID}
-        style={plainStyles}
-        numberOfLines={numberOfLines}
-        // @ts-ignore web only -prf
-        dataSet={WORD_WRAP}>
+      <Text selectable={selectable} testID={testID} style={[plainStyles, { fontSize }]} numberOfLines={numberOfLines}>
         {text}
       </Text>
-    )
+    );
   }
+  return (
+    <Text selectable={selectable} testID={testID} style={plainStyles} numberOfLines={numberOfLines}>
+      {text}
+    </Text>
+  );
+}
+
+const DEFAULT_DOMAIN = 'bsky.social';
+const TLD_REGEX = /.\w+$/;  // Regex to check for a valid top-level domain
 
-  const els = []
-  let key = 0
-  // N.B. must access segments via `richText.segments`, not via destructuring
+function buildTextElements(richText, disableLinks, enableTags, authorHandle, interactiveStyles, onLinkPress) {
+  let key = 0;
+  const elements = [];
   for (const segment of richText.segments()) {
-    const link = segment.link
-    const mention = segment.mention
-    const tag = segment.tag
-    if (
-      mention &&
-      AppBskyRichtextFacet.validateMention(mention).success &&
-      !disableLinks
-    ) {
-      els.push(
-        <ProfileHoverCard key={key} inline did={mention.did}>
-          <InlineLinkText
-            selectable={selectable}
-            to={`/profile/${mention.did}`}
-            style={interactiveStyles}
-            // @ts-ignore TODO
-            dataSet={WORD_WRAP}
-            onPress={onLinkPress}>
-            {segment.text}
-          </InlineLinkText>
-        </ProfileHoverCard>,
-      )
-    } else if (link && AppBskyRichtextFacet.validateLink(link).success) {
-      if (disableLinks) {
-        els.push(toShortUrl(segment.text))
-      } else {
-        els.push(
-          <InlineLinkText
-            selectable={selectable}
-            key={key}
-            to={link.uri}
-            style={interactiveStyles}
-            // @ts-ignore TODO
-            dataSet={WORD_WRAP}
-            shareOnLongPress
-            onPress={onLinkPress}>
-            {toShortUrl(segment.text)}
-          </InlineLinkText>,
-        )
+    const { link, mention, tag } = segment;
+
+    // Process mentions
+    if (mention) {
+      // Check if the mention contains a top-level domain to determine if it's a full handle
+      const hasTLD = TLD_REGEX.test(mention.did);
```

Comment:
Is the intention of this to allow the richtext did attribute stored in the post record to have a partial handle in it? I don't understand why. This change should be purely cosmetic in the UI.